ktor-server/ktor-server-core/jvm/src/io/ktor/server/http/content/SinglePageApplication.kt

```diff
@@ -26,45 +21,34 @@ import java.io.*
  * }
  * ```
  */
-public val SinglePageApplication: ApplicationPlugin<SPAConfig> = createApplicationPlugin(
-    "SinglePageApplication",
-    { SPAConfig() }
-) {
-    val defaultPage: String = pluginConfig.defaultPage
-    val applicationRoute: String = pluginConfig.applicationRoute
-    val filesPath: String = pluginConfig.filesPath
-    val ignoredFiles: MutableList<(String) -> Boolean> = pluginConfig.ignoredFiles
-    val usePackageNames: Boolean = pluginConfig.useResources
-
-    fun isUriStartWith(uri: String) =
-        uri.startsWith(applicationRoute) || uri.startsWith("/$applicationRoute")
+public fun Route.singlePageApplication(configBuilder: SPAConfig.() -> Unit = {}) {
+    val config = SPAConfig()
+    configBuilder.invoke(config)
 
     application.routing {
-        static(applicationRoute) {
-            if (usePackageNames) {
-                resources(filesPath)
-                defaultResource(defaultPage, filesPath)
-            } else {
-                staticRootFolder = File(filesPath)
-                files(".")
-                default(defaultPage)
+        static(config.applicationRoute) {
+            val shouldFileBeIgnored = { filePath: String ->
+                config.ignoredFiles.firstOrNull { it.invoke(filePath) } != null
             }
-        }
-    }
-
-    onCall { call ->
-        val requestUrl = call.request.uri
 
-        if (!isUriStartWith(requestUrl)) return@onCall
-
-        if (ignoredFiles.firstOrNull { it.invoke(requestUrl) } != null) {
-            call.respond(HttpStatusCode.Forbidden)
+            if (config.useResources) {
+                resourceWithDefault(
+                    config.filesPath,
+                    config.defaultPage,
+                    shouldFileBeIgnored
+                )
+                defaultResource(config.defaultPage, config.filesPath)
```

Comment:
Is a default fallback needed in presence of resource/filesWithDefault? Routes ending with a slash should be handled by the "not found" case.

Reply:
Thanks, you are right, I'will remove this line 👍